F16C triggers warning on ubuntu 17.04 with negative number.

F16C triggers a warning on ubuntu 17.04 with `gcc 6.3.0` on following line:

https://github.com/PetteriAimonen/libfixmath/blob/cbbfe85a4adb27394898600eea62afbacdb962f3/libfixmath/fix16.h#L310

When the sign is negative and only the value of `-FIXMATH_COMBINE_I_M((unsigned)( ( (i) * -1) ), m)` is of our interest, `FIXMATH_COMBINE_I_M` gets evaluated anyway which produces a warning because of the new -Wshift-negative-value
 
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65179

```
../libfixmath/fix16.h:274:9: error: left shift of negative value [-Werror=shift-negative-value]
         << 16 \
         ^
unittests.h:7:11: note: in definition of macro ‘TEST’
     if (!(x)) { \
           ^
../libfixmath/fix16.h:310:11: note: in expansion of macro ‘FIXMATH_COMBINE_I_M’
         : FIXMATH_COMBINE_I_M(i, m) \
           ^~~~~~~~~~~~~~~~~~~
fix16_macros_unittests.c:10:28: note: in expansion of macro ‘F16C’
   TEST(F16(i ## . ## m) == F16C(i,m)) \
                            ^~~~
fix16_macros_unittests.c:102:3: note: in expansion of macro ‘DO_TEST’
   DO_TEST(  -1,07737)
```

--- a/libfixmath/fix16.h
+++ b/libfixmath/fix16.h
@@ -307,7 +307,7 @@ extern fix16_t fix16_from_str(const char *buf);
     ( \
       (( #i[0] ) == '-') \
         ? -FIXMATH_COMBINE_I_M((unsigned)( ( (i) * -1) ), m) \
-        : FIXMATH_COMBINE_I_M(i, m) \
+        : FIXMATH_COMBINE_I_M((unsigned)i, m) \
     ) \
 )
 

